### DISCUSSION TITLE
AdminLTE with only an upgrade of Bootstrap

### DISCUSSION THREAD

**Discussion by craph** [ColorlibHQ/AdminLTE#4512] [Ideas]
Hello all,

I open this discussion regarding AdminLTE v4. Is it possible to plan a new version of AdminLTE with only the upgrade of the framework Bootstrap without adding new functionnality.

Is it what it's plan for the first release of the milestone v4.0.0 ? In the milestone I see many new feature, so may be is it possible to only have a version of AdminLTE with only the upgrade of Bootstrap as a first release and then plan other releases to add new features ?

Thank you very much.
Best regards,

**Comment by lenamtl** [comment]
Hi,

I'm looking for the same answer since a few months and still no updates ...
The todo list is the same, 22 task to do, we just don't know the status...
https://github.com/ColorlibHQ/AdminLTE/projects/13#card-78411124

I understand that this is free and done on spare time but I think we should have at least some updates...

Like you proposed the best option would be just to update to BS5 from the latest AdminLTE stable version.
They seems to follow BS5 idea by removing all jQuery scripts and replace by plain vanilla JS.
This can take time to develop, so this is maybe the reason why it is taking so much time.

I will still continue to use JQuery especially for Datatables and Select2..

I cannot wait any longer.
*In my case I'm still using old AminLTE v2 on Bootstrap 3  ...

So here my options:
1- Update my actual template Bootstrap 3  to just Bootstrap  5 (which can be faster but lack of new features)
2- Use AdminLTE v3  / Bootstrap 4 and just update it to Bootstrap 5 (I would have more new feature compare to solution 1)
3- Use commercial or other template based on Bootstrap 5 (this will cost money but this can save time and have more features and support)
4- Wait until AdminLTE v4 / Bootstrap 5 is ready and tested (this can take many months until the stable version is tested and has no bugs)
5- Create my own template 

Unless AdminLTE v4 / Bootstrap 5 is ready by the end of the week and has no bugs I will need a replacement solution....
I find it hard to evaluate the work, but I will need to take a decision by the end of the week.

I just saw that  AdminLTE v4 / BS5 is delayed until BS v5.30 stable is ready 
https://github.com/ColorlibHQ/AdminLTE/issues/3692#issue-891364273

Hope other dev will share their solution maybe some forks already exists.

**Reply by bhojkamal** [reply]
@lenamtl . I like option 2 or /and 3.  Please go for one of it.

**Reply by lenamtl** [reply]
Hi, 
Because my actual project is based on Adminlte v2 BS3
I have finally decided to update BS3 to BS5 and all related code, and update a few jQuery plugins.
This is the first step to do anyway.

Then when this done I will evaluate what need to be replaced (widget, menu).

**Comment by craph** [comment]
@danny007in Please can we have an update about the branch v4-dev ? can we help ?

**Reply by danny007in** [reply]
https://github.com/ColorlibHQ/AdminLTE/projects/13 please check it, and try to contribute

**Comment by lenamtl** [comment]
Hi @craph

There is no much difference between BS4 and BS5
https://getbootstrap.com/docs/5.0/migration/

I'm testing AdminLte v3 / BS4 **with BS5** so I have only **replaced the Bootstrap directory** and **not changing anything to the code** for the test.

Until now only a few problem found : Tab, modal, close alert, button dropdown, toggle button

To fix toggle code that use toggle : Tab, Button dropdown, Accordion, Toogle
change 
`data-toggle `
by
`data-bs-toggle`

Alert
-Close Alert 

change
`data-dismiss="alert"`
by
`data-bs-dismiss="alert"`

Other that need to be adjusted:
Carousel
-Slide

-Bootstrap Modal

**Reply by bhojkamal** [reply]
@lenamtl , Yes, not much difference from BS4 to BS5, is just changing few thing, it would work. `data-dismiss="modal"` to `data-bs-dismiss="modal"`. Basically, if there is `data-something` we need make it `data-bs-something`. all `text-right` to `text-end`, basically `-left` to `start`,  `-right` to `-end`. `pl, ml` to `ps, ms` `pr, mr` to `pe, me`. and break point `xxl` is added. I've used adminlte 3 and added BS5 CDN and it is working without breaking. So, it would be nice if the admin lte itself upgraded to BS5.

**Reply by lenamtl** [reply]
I'm experiencing a random issue when clicking  on button (menu or other) or when clicking on toggle button there is a delay.
I don't know what cause the issues, no error in JS, I will continue the investigation...

**Reply by danny007in** [reply]
> I'm experiencing a random issue when clicking on button (menu or other) or when clicking on toggle button there is a delay. I don't know what cause the issues, no error in JS, I will continue the investigation...

please create new issue

**Reply by lenamtl** [reply]
@danny007in this is not related to any version v3 BS4 or v4 BS, it is related to v3 BS4 using BS5 instead of BS4.
I'm testing this fork.

I would prefer to use  AdminLte v4 BS5, but it is not ready and there is no update.

So I'm trying to see if using AdminLte v3 and only replacing BS4 with BS5 can work fine or not.
The only major issue I have after fixing code to use BS5 classes is the click delay (no JS error just delay +-5 sec).
I'm testing localy. I guess the issues probably with the adminlte.js & BS5 JS.
I'm not using AdminLte v3 so maybe the problem exist I will check and I will open a new issue if this occurred on original  v3 .

I'm doing many tests with this fork to see if this is possible to use this fork or not in production,
I will remove all plugins that I don't use to see if this click issues is still present or not.

I have to take a decision by the end of the week, I need to update my actual project who still use v2.

**Reply by lenamtl** [reply]
@danny007in I have reproduced the problem so I have opened a new issue for this 
https://github.com/ColorlibHQ/AdminLTE/issues/4527

**Comment by JaumeAG** [comment]
What steps must be followed to update from adminlte v2 and BTS 3 to adminlte 3 and BTS 4?
